stdlib/public/Python/NumpyConversion.swift

```diff
@@ -0,0 +1,180 @@
+//===-- NumpyConversion.swift ---------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the `NumpyArrayConvertible` protocol for bridging
+// `numpy.ndarray`.
+//
+//===----------------------------------------------------------------------===//
+
+/// A type that can be initialized from a `numpy.ndarray` instance represented
+// as a `PyValue`.
+public protocol NumpyArrayConvertible {
+  init?(numpyArray: PyValue)
+}
+
+/// A type that is compatible with a NumPy scalar `dtype`.
+public protocol NumpyScalarCompatible {
+  static func isCompatibleNumpyScalarType(_ dtype: PyValue) -> Bool
```

Comment:
I was talking about a random hypothetical name yesterday :) There can be a better name for this.

How about `isCompatible(withNumpyScalarType:)`?